tests/scenario/test_deploy.py

```diff
@@ -47,15 +47,30 @@ def patch_urllib_request():
         "metrics-generator",
     ],
 )
+@patch.object(JujuTopology, "from_charm")
 @patch.object(ClusterRequirer, "get_worker_config", MagicMock(return_value={"config": "config"}))
-def test_pebble_ready_plan(ctx, role):
+def test_pebble_ready_plan(topology_mock, ctx, role):
+    charm_topo = JujuTopology(
+        model="test",
+        model_uuid="00000000-0000-4000-8000-000000000000",
+        application="worker",
+        unit="worker/0",
+        charm_name="tempo",
+    )
+    topology_mock.return_value = charm_topo
+
     expected_plan = {
         "services": {
             "tempo": {
                 "override": "replace",
                 "summary": "tempo worker process",
                 "command": f"/bin/tempo -config.file=/etc/worker/config.yaml -target {role if role != 'all' else 'scalable-single-binary'}",
                 "startup": "enabled",
+                "environment": {
```

Comment:
also I would add a test that verifies that the env isn't there if we don't have a jaeger endpoint enabled for whatever reason